release-notes-1.2.0.md

```diff
@@ -0,0 +1,751 @@
+
+```ballerina
+import ballerina/lang.'xml as xmllib;
+
+public function main() {
+     xmllib:Element element = <xmllib:Element> xml `<elem> hello </elem>`;
+     xml children = element.getChildren();
+   }
+```
+
+#### The `lang.boolean` module
+
+The newly-added lang library module for the `boolean` basic type contains a function for parsing `string` values to `boolean` values. It accepts `”true”` or `”false”` in any combination of lower/upper case as well as `”1”` and `”0”`, which evaluates to `true` and `false` respectively. An error is returned for any other `string` value.
+
+```ballerina
+import ballerina/lang.'boolean;
+
+public function main() {
+    boolean|error b = 'boolean:fromString("true");
+}
+```
+
+#### Enhanced `lang.int`, `lang.string`, `lang.array`, and `lang.map` modules
+
+The above-mentioned lang library modules have been enhanced by adding new API elements (e.g., constants and functions). 
+
+- `lang.int` module - Constants added: for max/min values of the `int` type 
+- `lang.string` module - New functions added: `lastIndexOf()`, `equalsIgnoreCaseAscii()`, `toCodePointInt()`, `fromCodePointInt()`
+- `lang.array` module - New functions added: `lastIndexOf()`, `toStream()`
+- `lang.map` module - New functions added: `toArray()`, `removeIfHasKey()`
+
+
+### Backward incompatible improvements and bug fixes
+- Listener variables are now final.
+- The syntax for Unicode escapes in strings has changed from `\u[CodePoint]` to `\u{CodePoint}` so as to align with ECMAScript. Although this is an incompatible change, the previous syntax was not implemented.
+- The semantics of the `lock` statement has been changed. Now, it acquires a single, program-wide, recursive mutex before executing a lock statement and releases the mutex after completing the execution of the lock statement.
+- When a list constructor or a mapping constructor is used without a contextually-expected type, now a tuple or a record type is inferred rather than an array or a map type.
+- The `x@` syntax  for accessing the attributes of an XML element has been removed.
+- Now, member value assignment for a dynamic list adds filler values if the required members are not already filled and the relevant member type has a filler value.
+- Functions with the same name were previously allowed in both the module and the module's tests. This is now disallowed and results in a compilation error.
+- Now, out of range index access of `xml` sequences result in empty sequences rather than a panic. Using the `xml:get()` langlib method with an out of range index results in a panic.
+
+## Runtime
+
+### Compilation time improvement
+
+Major improvements to the compilation (code-generation) time are implemented. The compiler backend, which generates the Java bytecode for Ballerina programs is now rewritten to improve the overall compilation time. This change has specifically reduced the total time it takes for Java bytecode generation.
```

Comment:
All 3 sentences pretty much say the same thing.
```suggestion
The compiler backend, which generates the Java bytecode for Ballerina programs was rewritten to reduce the time taken for Java bytecode generation. This resulted in a significant reduction in overall compilation time.
```

Reply:
+1